src/java/PermutationManager.java

```diff
@@ -14,39 +14,41 @@
 public class PermutationManager
 {
 
-    private static PermutationManager singleton;
-    private static Integer lock;
+    private static PermutationManager singleton; // declares that there can only ever be one PermutationManager instance at a time?
+    private static Integer lock; // declared for the purposes for the synchronized block later?
```

Comment:
Yes it can be anything really, doesn't need to be an integer. The == operator is used on the argument to "synchronized" to determine if two synchronized blocks should block each other (synchronously).